In [1]:
import pandas as pd
import numpy as np

In [2]:
%ls datasets/

adult.csv                  census.csv     nhanes_2015_2016.csv
airline_data.csv           coches.csv     pokemon.csv
antropometria00.csv        cwurData.csv   reg_data.csv
antropometria-dataset.csv  data.csv       sample_pivot.csv
Book3.csv                  headbrain.csv  tips.csv
Canada.xlsx                mic2.xlsx      weight-height.csv
Cartwheeldata.csv          movies.csv


In [3]:
pip install openpyxl

Note: you may need to restart the kernel to use updated packages.


In [4]:
df = pd.read_excel('datasets/Canada.xlsx',
                   sheet_name = 'Canada by Citizenship',
                  skiprows = range(20), 
                  skipfooter = 2,
                  engine = 'openpyxl'
                  )

In [5]:
pip install xlrd

Note: you may need to restart the kernel to use updated packages.


In [6]:
df

Type    Coverage          OdName  AREA AreaName   REG  \
0    Immigrants  Foreigners     Afghanistan   935     Asia  5501   
1    Immigrants  Foreigners         Albania   908   Europe   925   
2    Immigrants  Foreigners         Algeria   903   Africa   912   
3    Immigrants  Foreigners  American Samoa   909  Oceania   957   
4    Immigrants  Foreigners         Andorra   908   Europe   925   
..          ...         ...             ...   ...      ...   ...   
190  Immigrants  Foreigners        Viet Nam   935     Asia   920   
191  Immigrants  Foreigners  Western Sahara   903   Africa   912   
192  Immigrants  Foreigners           Yemen   935     Asia   922   
193  Immigrants  Foreigners          Zambia   903   Africa   910   
194  Immigrants  Foreigners        Zimbabwe   903   Africa   910   

                RegName  DEV             DevName  1980  ...  2004  2005  2006  \
0         Southern Asia  902  Developing regions    16  ...  2978  3436  3009   
1       Southern Europe  901   Developed regions     1  ...  1450  1223   856   
2       Northern Africa  902  Developing regions    80  ...  3616  3626  4807   
3             Polynesia  902  Developing regions     0  ...     0     0     1   
4       Southern Europe  901   Developed regions     0  ...     0     0     1   
..                  ...  ...                 ...   ...  ...   ...   ...   ...   
190  South-Eastern Asia  902  Developing regions  1191  ...  1816  1852  3153   
191     Northern Africa  902  Developing regions     0  ...     0     0     1   
192        Western Asia  902  Developing regions     1  ...   124   161   140   
193      Eastern Africa  902  Developing regions    11  ...    56    91    77   
194      Eastern Africa  902  Developing regions    72  ...  1450   615   454   

     2007  2008  2009  2010  2011  2012  2013  
0    2652  2111  1746  1758  2203  2635  2004  
1     702   560   716   561   539   620   603  
2    3623  4005  5393  4752  4325  3774  4331  
3       0     0     0     0     0     0     0  
4       1     0     0     0     0     1     1  
..    ...   ...   ...   ...   ...   ...   ...  
190  2574  1784  2171  1942  1723  1731  2112  
191     0     0     0     0     0     0     0  
192   122   133   128   211   160   174   217  
193    71    64    60   102    69    46    59  
194   663   611   508   494   434   437   407  

[195 rows x 43 columns]

In [7]:
%ls

clase2.ipynb  datasets/  README.md  requirements.txt


In [8]:
df.head()

Type    Coverage          OdName  AREA AreaName   REG  \
0  Immigrants  Foreigners     Afghanistan   935     Asia  5501   
1  Immigrants  Foreigners         Albania   908   Europe   925   
2  Immigrants  Foreigners         Algeria   903   Africa   912   
3  Immigrants  Foreigners  American Samoa   909  Oceania   957   
4  Immigrants  Foreigners         Andorra   908   Europe   925   

           RegName  DEV             DevName  1980  ...  2004  2005  2006  \
0    Southern Asia  902  Developing regions    16  ...  2978  3436  3009   
1  Southern Europe  901   Developed regions     1  ...  1450  1223   856   
2  Northern Africa  902  Developing regions    80  ...  3616  3626  4807   
3        Polynesia  902  Developing regions     0  ...     0     0     1   
4  Southern Europe  901   Developed regions     0  ...     0     0     1   

   2007  2008  2009  2010  2011  2012  2013  
0  2652  2111  1746  1758  2203  2635  2004  
1   702   560   716   561   539   620   603  
2  3623  4005  5393  4752  4325  3774  4331  
3     0     0     0     0     0     0     0  
4     1     0     0     0     0     1     1  

[5 rows x 43 columns]

In [9]:
df.columns

Index([    'Type', 'Coverage',   'OdName',     'AREA', 'AreaName',      'REG',
        'RegName',      'DEV',  'DevName',       1980,       1981,       1982,
             1983,       1984,       1985,       1986,       1987,       1988,
             1989,       1990,       1991,       1992,       1993,       1994,
             1995,       1996,       1997,       1998,       1999,       2000,
             2001,       2002,       2003,       2004,       2005,       2006,
             2007,       2008,       2009,       2010,       2011,       2012,
             2013],
      dtype='object')

In [10]:
df.shape
# to see the size of the information

(195, 43)

In [11]:
type(df.columns)
#to se the type of a df

pandas.core.indexes.base.Index

In [12]:
df.columns.tolist()
# to see the columns on a list form

['Type',
 'Coverage',
 'OdName',
 'AREA',
 'AreaName',
 'REG',
 'RegName',
 'DEV',
 'DevName',
 1980,
 1981,
 1982,
 1983,
 1984,
 1985,
 1986,
 1987,
 1988,
 1989,
 1990,
 1991,
 1992,
 1993,
 1994,
 1995,
 1996,
 1997,
 1998,
 1999,
 2000,
 2001,
 2002,
 2003,
 2004,
 2005,
 2006,
 2007,
 2008,
 2009,
 2010,
 2011,
 2012,
 2013]

In [13]:
df.head(2)
#to see the amount of rows that you want

Type    Coverage       OdName  AREA AreaName   REG          RegName  \
0  Immigrants  Foreigners  Afghanistan   935     Asia  5501    Southern Asia   
1  Immigrants  Foreigners      Albania   908   Europe   925  Southern Europe   

   DEV             DevName  1980  ...  2004  2005  2006  2007  2008  2009  \
0  902  Developing regions    16  ...  2978  3436  3009  2652  2111  1746   
1  901   Developed regions     1  ...  1450  1223   856   702   560   716   

   2010  2011  2012  2013  
0  1758  2203  2635  2004  
1   561   539   620   603  

[2 rows x 43 columns]

In [14]:
df.drop(['Type', 'Coverage',     'AREA',    'REG',
        'RegName',      'DevName',],
        axis = 1,
        inplace = True
        )
#axis = 1 column, axis = 0 Row
#inplace = true, delets it from the source

In [15]:
df.rename(columns={'OdName' : 'Country', 'AreaName' : 'Continent',},
          inplace = True
         )
#Change the name of the column that you type
df.head(2)

Country Continent  DEV  1980  1981  1982  1983  1984  1985  1986  ...  \
0  Afghanistan      Asia  902    16    39    39    47    71   340   496  ...   
1      Albania    Europe  901     1     0     0     0     0     0     1  ...   

   2004  2005  2006  2007  2008  2009  2010  2011  2012  2013  
0  2978  3436  3009  2652  2111  1746  1758  2203  2635  2004  
1  1450  1223   856   702   560   716   561   539   620   603  

[2 rows x 37 columns]

#### iloc[RowStart:RowEnd_,_ColumnStart:ColumnEnd]

In [16]:
#df['Total'] = df.iloc[:,3:].sum(axis = 1)
df['Total'] = df.iloc[:,3:].sum(axis=1)

In [17]:
df.head()

Country Continent  DEV  1980  1981  1982  1983  1984  1985  1986  \
0     Afghanistan      Asia  902    16    39    39    47    71   340   496   
1         Albania    Europe  901     1     0     0     0     0     0     1   
2         Algeria    Africa  902    80    67    71    69    63    44    69   
3  American Samoa   Oceania  902     0     1     0     0     0     0     0   
4         Andorra    Europe  901     0     0     0     0     0     0     2   

   ...  2005  2006  2007  2008  2009  2010  2011  2012  2013  Total  
0  ...  3436  3009  2652  2111  1746  1758  2203  2635  2004  58639  
1  ...  1223   856   702   560   716   561   539   620   603  15699  
2  ...  3626  4807  3623  4005  5393  4752  4325  3774  4331  69439  
3  ...     0     1     0     0     0     0     0     0     0      6  
4  ...     0     1     1     0     0     0     0     1     1     15  

[5 rows x 38 columns]

In [18]:
df.columns

Index([  'Country', 'Continent',       'DEV',        1980,        1981,
              1982,        1983,        1984,        1985,        1986,
              1987,        1988,        1989,        1990,        1991,
              1992,        1993,        1994,        1995,        1996,
              1997,        1998,        1999,        2000,        2001,
              2002,        2003,        2004,        2005,        2006,
              2007,        2008,        2009,        2010,        2011,
              2012,        2013,     'Total'],
      dtype='object')

In [19]:
#This will help you to find a specific point on your data (the 3 lines below show you how)
df.set_index('Country', 
             inplace = True)
df

Continent  DEV  1980  1981  1982  1983  1984  1985  1986  1987  \
Country                                                                         
Afghanistan         Asia  902    16    39    39    47    71   340   496   741   
Albania           Europe  901     1     0     0     0     0     0     1     2   
Algeria           Africa  902    80    67    71    69    63    44    69   132   
American Samoa   Oceania  902     0     1     0     0     0     0     0     1   
Andorra           Europe  901     0     0     0     0     0     0     2     0   
...                  ...  ...   ...   ...   ...   ...   ...   ...   ...   ...   
Viet Nam            Asia  902  1191  1829  2162  3404  7583  5907  2741  1406   
Western Sahara    Africa  902     0     0     0     0     0     0     0     0   
Yemen               Asia  902     1     2     1     6     0    18     7    12   
Zambia            Africa  902    11    17    11     7    16     9    15    23   
Zimbabwe          Africa  902    72   114   102    44    32    29    43    68   

                ...  2005  2006  2007  2008  2009  2010  2011  2012  2013  \
Country         ...                                                         
Afghanistan     ...  3436  3009  2652  2111  1746  1758  2203  2635  2004   
Albania         ...  1223   856   702   560   716   561   539   620   603   
Algeria         ...  3626  4807  3623  4005  5393  4752  4325  3774  4331   
American Samoa  ...     0     1     0     0     0     0     0     0     0   
Andorra         ...     0     1     1     0     0     0     0     1     1   
...             ...   ...   ...   ...   ...   ...   ...   ...   ...   ...   
Viet Nam        ...  1852  3153  2574  1784  2171  1942  1723  1731  2112   
Western Sahara  ...     0     1     0     0     0     0     0     0     0   
Yemen           ...   161   140   122   133   128   211   160   174   217   
Zambia          ...    91    77    71    64    60   102    69    46    59   
Zimbabwe        ...   615   454   663   611   508   494   434   437   407   

                Total  
Country                
Afghanistan     58639  
Albania         15699  
Algeria         69439  
American Samoa      6  
Andorra            15  
...               ...  
Viet Nam        97146  
Western Sahara      2  
Yemen            2985  
Zambia           1677  
Zimbabwe         8598  

[195 rows x 37 columns]

In [20]:
df.index.name = None

In [21]:
df.loc['Japan']

Continent     Asia
DEV            901
1980           701
1981           756
1982           598
1983           309
1984           246
1985           198
1986           248
1987           422
1988           324
1989           494
1990           379
1991           506
1992           605
1993           907
1994           956
1995           826
1996           994
1997           924
1998           897
1999          1083
2000          1010
2001          1092
2002           806
2003           817
2004           973
2005          1067
2006          1212
2007          1250
2008          1284
2009          1194
2010          1168
2011          1265
2012          1214
2013           982
Total        27707
Name: Japan, dtype: object

### Conclusion:

A total of 27707 people emmigrated to Canada from Japan during 1980 to 2013